xtask/src/snapshot.rs

```diff
@@ -25,15 +28,19 @@ pub(crate) fn all_snapshot_tests() -> Vec<&'static str> {
         "hints",
         "hints_inner",
         "dbg",
-        "net",
         "panic_info",
     ];
     const NIGHTLY_SNAPSHOT_TESTS: &[&str] = &["alloc"];
+    const POST_MSRV_SNAPSHOT_TESTS: &[&str] = &["net"];
```

Comment:
It's unclear to me what "post msrv snapshot tests" are. If we raise the msrv should this change?

Reply:
Yes. We have three brackets of tests
1. tests that work on the msrv, stable and nightly
2. tests that work on stable and nightly
3. tests that work on nightly only

Regarding 2: The `default` defmt features support the msrv, but some of the optional features require a newer version, e.g. `ip_in_core`. These tests are the ones in `POST_MSRV_TESTS`. Maybe `OPTIONAL_FEATURE_TESTS` is better?

Reply:
So `POST_MSRV` means "any version above our MSRV"? If so, a little comment would be really nice:

```rust
// Tests which run on all Rustc versions later than our MSRV
const POST_MSRV_SNAPSHOT_TESTS: &[&str] = &["net"];
```

Reply:
Well it's not all versions. It's some version in `MSRV..=STABLE`.